### DISCUSSION TITLE
Issues building with RSL support

### DISCUSSION THREAD

**Discussion by flappity** [ARM-DOE/pyart#1247] [Q&A] [answered]
Hi!  I've been trying to get Py-ART installed with support for RSL (this is within an Ubuntu 20.04 WSL2 instance on win10).  I have tried a few different ways:

I've built RSL 1.5.0 from source into the default location, and installed Py-ART afterwards (without setting RSL_PATH), assuming it would see it in its default location and build with it.  No luck

I've tried installing 1.5.0 into the environment itself, pointing RSL_PATH at /home/user/anaconda3/env/pyart_rsl/.  Still seems to not pick up RSL.

Next attempt, I assumed maybe I had to build Py-ART myself with `python setup.py install`, but that introduced another issue I didn't know how to fix (circular dependency when importing pyart), and by all accounts it seems that the conda/pip installs should work fine anyways.

So on my next try, I tried installing RSL using `conda install -c conda-forge trmm_rsl`, and then installing pyart (with RSL_PATH set to /home/user/anaconda3/envs/pyart_rsl/) -- this did not work either.

Looking into some of the Cython logs, I do see references to whatever location I have set in RSL_PATH, so it seems to be reading it?  It's just seemingly refusing to build with support for it. Anytime I try to use something that requires RSL, I get the error `pyart.exceptions.MissingOptionalDependency: Py-ART must be build with support for TRMM RSL to use the dealias_fourdd function.` (or whatever function I'm using)

I feel like I'm missing out on something completely obvious here, but the only other references to this problem I can find end with something along the lines of "Looks like I got it! Thanks" without a solution given.

Any chance you can tell me what key thing I might be missing here? 

Thanks!

**Comment by zssherman** [comment, marked as answer]
Hi @flappity, Max and I been trying to figure this out for some time. conda-forge did changes to their build for the new MacOS chips, we tried to do a release for Py-ART for support of these chips and needed a release for trmm_rsl as well. trmm_rsl was last release in 2017 for conda-forge. We tried doing a release and passed for the conda-forge testing, but as you mentioned, we couldn't get the path to be found from linux systems as well. MacOS (non-new chips) was working. We tried changing the export path and more and could not figure it out.

We then tried to to a 1.50 release, but were having compiler issues., paths not being found within trmm_rsl, And dependencies not being found etc.

Scott, Max, and I then were discussing what rsl is being used for by the community. Could we encourage using region dealiaser instead of the 4DD dealiaser or poll and see what RSL is being used for (4DD or reader) and how many indidivuals use it. Still in discussion, and we are still seeing what else we can try as Max and I don't have a strong C knowledge to fix trmm_rsl itself, if it needs fixing. Long story short we been trying to figure out why trmm_rsl is not working and how to fix it and other paths forward.

**Reply by flappity** [reply]
Ahh, I understand -- glad to know I wasn't going crazy.  RSL probably isn't that important to have, as I believe LROSE has RadxDealias that also uses FourDD as well as RadxConvert which I can use liberally to get files into Py-ART readable format.  It was mostly just a convenience thing (along with the fact that I couldn't figure out why it was inexplicably failing to work).
